docs/content/en/docs/clustermgmt/cluster-upgrades/vsphere-and-cloudstack-upgrades.md

```diff
@@ -15,6 +15,8 @@ The upgrade command also upgrades core components of EKS Anywhere and lets the u
 
 **Upgrades should never be run from ephemeral nodes (short-lived systems that spin up and down on a regular basis). If an upgrade fails, it is very important not to delete the Docker containers running the KinD bootstrap cluster. During an upgrade, the bootstrap cluster contains critical EKS Anywhere components. If it is deleted after a failed upgrade, they cannot be recovered.**
 
+**Please make sure to have sufficient available IP addresses in your DHCP pool from VMs counts and maxSurge config. For create operation, each VM needs 1 IP. For upgrade and scale operation, control plane/works needs 1 extra IP, each external ETCD needs 1 extra IP, plus IPs equal to the number used for maxSurge. After calculating the required IPs, please make sure your environment has enough available IPs before upgrade operation. For example, if you need to create a cluster with 1 control plane node, 2 worker node and 3 external ETCD nodes, you'll need at least 6 (1+2+3) IPs available. To upgrade the same cluster with maxSurge set to 2, you'll need 6 (1+3+2) available IPs.
+
```

Comment:
For the upgrade, is that 6 additional IPs?

Reply:
i think it worths mention that because we use rolling upgrade as upgrade strategy, that's why extra one VM is needed for machine rollout. Also explain why "each external ETCD needs 1 extra IP" -- etcdadm controller needs to create all the new etcd machines before removing the old etcd machines.

Reply:
also just my own preference. Can we have like a few examples in the end w detailed calculations?

Format be like: a paragraph explaining the recommendation, rolling upgrade strategy, how things are calculated. Then 2-3 examples: 1) unstacked 3+3+3 + max surge is default (0) 2) stacked etcd case 3) ...

Reply:
Agree with above comments,  plus few minor wording tweaks
control plane/works needs 1 extra IP -> control plane and workers need 1 extra IP (total not per node)
 each external ETCD needs 1 extra IP -> each external ETCD node needs 1 extra IP address (ex: 3 ETCD nodes would require 3 more IP addresses)
least 6 (1+2+3) IPs available -> least 6 (1+2+3) available IPs